app/controllers/concerns/controllers/locale.rb

```diff
@@ -2,22 +2,27 @@
 
 # License: AGPL-3.0-or-later WITH WTO-AP-3.0-or-later
 # Full license explanation at https://github.com/houdiniproject/houdini/blob/main/LICENSE
-# rubocop:disable Style/ConditionalAssignment
 module Controllers::Locale
 	extend ActiveSupport::Concern
 
 	included do
-		before_action :set_locale
+		around_action :switch_locale
 
 		private
 
-		def set_locale
-			if params[:locale] && Houdini.intl.available_locales.include?(params[:locale])
-				I18n.locale = params[:locale]
-			else
-				I18n.locale = Houdini.intl.language
-			end
+		def switch_locale(&action)
+			locale = if Houdini.intl.available_locales.include?(params[:locale])
+													params[:locale]
+												else
+													http_accept_language.compatible_language_from(Houdini.intl.available_locales)
+												end
+			locale = adequate_locale_to_transifex_syntax(locale)
+			logger.debug "* Locale set to '#{locale}'"
+			I18n.with_locale(locale, &action)
+		end
+
+		def adequate_locale_to_transifex_syntax(locale)
```

Comment:
Although the default keys from locales that include variants such as Brazilian Portuguese is to use a dash (`-`) to determine the variant (as in `pt-BR`), Transifex uses underscore.

Reply:
Is it possible to change the underscores from Transifex to dashes before merging? I think it would be clearer if it worked that way.

Reply:
I couldn't find this option anywhere :(